### DISCUSSION TITLE
How are test cases in classes named "*Tests.java" executed?

### DISCUSSION THREAD

**Discussion by jonbartels** [nextgenhealthcare/connect#5417] [Q&A]
I was playing with Mirth Connect over the weekend and it looks like test classes which match the pattern `*.Test` are executed but not `*.Tests` . I am wondering if this is intentional and if there is a secondary test mechanism one should use.

My goal is to learn how the MC test suites are run so that when I file bug reports I can also write test cases for them.

What I see is:
* The [build.xml from the `server` module](https://github.com/nextgenhealthcare/connect/blob/442b0895ab988071a294b5fda68b59474a4de8b7/server/build.xml#L1244-L1248) only includes classes matching `*Test.class`
* There are test cases which a [`Tests` naming convention](https://github.com/nextgenhealthcare/connect/blob/e66d1122dbeebfcf5a8939beb93d720b2d5071fb/server/test/com/mirth/connect/plugins/datatypes/hl7v2/HL7SerializerTests.java) 
* When I run the `ant` build I see tests being executed but I cannot find reports for the `Tests` classes. This leads me to believe they are not executed.
* I found [`mirth-ci.xml`](https://github.com/nextgenhealthcare/connect/blob/development/server/mirth-ci.xml) which seems to be for CI/CD but it just calls the main build file
* The tests were renamed [in these commits for MIRTH-4120](https://github.com/nextgenhealthcare/connect/search?q=MIRTH-4120&type=commits) . 

How are the `Tests` classes meant to be run?

**Comment by jonbartels** [comment]
Per a user in Slack, I added `<include name="**/*Tests.class" />` to the `test-run` step.

I now see the `*Tests` classes being executed. This does prove that the tests are executable. I'm looking at the test reports to see if they are successful.

Even if they are executable, my key question still needs an answer: How are the `Tests` classes meant to be run?

**Reply by tonygermano** [reply]
I don't think they are meant to be run. This one is certainly not passing as it looks like it requires the test to be run on Windows with a certain set of dicom files present on the root of the C: drive.

https://github.com/nextgenhealthcare/connect/blob/4.1.1/server/test/com/mirth/connect/plugins/datatypes/dicom/DICOMTests.java
